### PR TITLE
EID-1321: Pass X509Certificate as a String

### PR DESCRIPTION
**Pull request by CharlesIC**
We can create the cert and credential from the string in the Translator

### CODE REVIEW

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserRequest.java [hunk 1]
```diff
@@ -7,9 +7,6 @@
     @JsonProperty
     private String authnRequest;
 
-    public EidasSamlParserRequest() {
```
Comment (Wynndow) [inline proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserRequest.java]: I think the default constructor is needed for Jackson to serialise the DTO.

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserResponse.java [hunk 2]
```diff
@@ -20,18 +17,13 @@
...
-    @JsonSerialize(using = X509CertificateSerializer.class)
-    @JsonDeserialize(using = X509CertificateDeserializer.class)
-    private X509Certificate connectorEncryptionPublicCertificate;
+    private String connectorEncryptionPublicCertificate;
 
     @JsonProperty
     @NotBlank
     private String destination;
 
-    private EidasSamlParserResponse() {
```
Comment (Wynndow) [inline proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserResponse.java]: Same as above, I think it's needed for serialisation.

**Comment by Wynndow** [comment]
Why rather a string over the serialized object?

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserRequest.java (hunk 1 above)
Comment (CharlesIC) [reply proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserRequest.java]: I think you're right!

File: proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserResponse.java (hunk 2 above)
Comment (CharlesIC) [reply proxy-node-shared/src/main/java/uk/gov/ida/notification/contracts/EidasSamlParserResponse.java]: Restored 👍

**Pull request merged**